monstache.go

```diff
@@ -4673,7 +4682,7 @@ func (ic *indexClient) eventLoop() {
 }
 
 func (ic *indexClient) startIndex() {
-	for i := 0; i < 5; i++ {
+	for i := 0; i < ic.config.MapperPluginThreads; i++ {
```

Comment:
This loop eventually calls the Map function in the plugin. It starts multiple Goroutines to process documents from Mongo. If the setting is hardcoded to 5 that means only 5 plugins can be executed at the same time therefore introducing an artificial bottleneck. This is not an issue if there is no plugin configured or if the plugin is just transforming documents and executes really fast.